### PR TITLE
Allow blocking users

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add flow to test confirming using token
- Add User::BlockDomain model
- Users with blocked email domain are not confirmed
- Allow disabling users

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -274,4 +274,8 @@ def send_devise_notification(notification, *args)
 
   def may_create_profile? = reputation >= User::Profile::MIN_REPUTATION
   def profile? = profile.present?
+
+  def confirmed? = super && !disabled? && !blocked?
```
Comment (ErikSchierboom) [inline app/models/user.rb:278]: I've ordered the checks to have the non-expensive checks first

File: app/models/user/block_domain.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class User::BlockDomain < ApplicationRecord
+  def self.blocked?(user)
+    domain = Mail::Address.new(user.email).domain
```
Comment (ErikSchierboom) [inline app/models/user/block_domain.rb:3]: Is it possible for a user to not have an email?

File: app/models/user.rb (hunk 1 above)
Comment (iHiD) [reply app/models/user.rb:278]: Nice.

File: app/models/user/block_domain.rb (hunk 2 above)
Comment (iHiD) [reply app/models/user/block_domain.rb:3]: No.

**Pull request merged**